File: src/main/java/xyz/emirdev/echogen/events/PlayerQuitListener.java
package xyz.emirdev.echogen.events;

import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;
import xyz.emirdev.echogen.Echogen;
import xyz.emirdev.echogen.utils.Utils;

public class PlayerQuitListener implements Listener {

    @EventHandler()
    public void onQuit(PlayerQuitEvent event) {
        String quitMessage = Echogen.get().getPluginConfig().getRootNode().node("chat", "messages", "leave").getString();
        if (quitMessage == null) return;

        if (quitMessage.isEmpty()) {
            event.quitMessage(null);
            return;
        }

        Player player = event.getPlayer();

        Utils.formatMessage(quitMessage,
                Placeholder.parsed("name", player.getName())
        );
    }
}

File: src/main/kotlin/cat/emir/echogen/events/PlayerQuitListener.kt
package cat.emir.echogen.events

import me.clip.placeholderapi.PlaceholderAPI
import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent
import cat.emir.echogen.Echogen
import cat.emir.echogen.toComponent

class PlayerQuitListener(val plugin: Echogen) : Listener {

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        var quitMessage = plugin.config.rootNode.node("chat", "messages", "leave").string ?: return

        if (quitMessage.isEmpty()) {
            event.quitMessage(null)
            return
        }

        val player = event.player

        if (plugin.isPAPIEnabled) {
            quitMessage = PlaceholderAPI.setPlaceholders(player, quitMessage)
        }

        event.quitMessage(quitMessage.toComponent(
                Placeholder.parsed("name", player.name)
        ))
    }
}
